File: src/main/java/com/eh2/noframe/infrastructure/db/H2Init.java
package com.eh2.noframe.infrastructure.db;

import java.net.URISyntaxException;
import java.net.URL;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;

public class H2Init {
	public static void main(String[] args) throws URISyntaxException {
		URL scriptUrl = H2Init.class
		        .getClassLoader()
		        .getResource("script.sql");

		Path path = Paths.get(scriptUrl.toURI());
		
		if (scriptUrl == null) {
			System.out.println("Script file not found!");
			return;
		}
		try (Connection conn = DBConnection.getConnection()) {
			try (Statement st = conn.createStatement()) {
				st.execute("RUNSCRIPT FROM '" +path.toString() + "'");
			}
		} catch (SQLException e) {
			throw new RuntimeException(e);
		}
	}
}

File: src/main/kotlin/com/eh2/noframe/infrastructure/db/H2Init.kt
package com.eh2.noframe.infrastructure.db

import java.net.URISyntaxException
import java.net.URL
import java.nio.file.Path
import java.nio.file.Paths
import java.sql.SQLException

object H2Init {

    @JvmStatic
    @Throws(URISyntaxException::class)
    fun main(args: Array<String>) {

        val scriptUrl: URL = H2Init::class.java
            .classLoader
            .getResource("script.sql")
            ?: throw IllegalArgumentException("script.sql not found")

        val path: Path = Paths.get(scriptUrl.toURI())

        try {
            DBConnection.getConnection().use { conn ->
                conn.createStatement().use { st ->
                    st.execute("RUNSCRIPT FROM '${path}'")
                }
            }
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }
}